qtwidgets/scan_plotwidget.py

```diff
@@ -0,0 +1,322 @@
+        self.vline.setPos(pos[0])
+        self.hline.setPos(pos[1])
+        self.sigCrosshairDraggedPosChanged.emit(QtCore.QPointF(pos[0], pos[1]))
+        return
+
+    def toggle_crosshair(self, enable, movable=True):
+        """
+
+        @param bool enable:
+        @param bool movable:
+        """
+        if not isinstance(enable, bool):
+            raise TypeError('Positional argument "enable" must be bool type.')
+        if not isinstance(movable, bool):
+            raise TypeError('Optional argument "movable" must be bool type.')
+
+        self.crosshair_movable = movable
+
+        if enable:
+            if self.vline not in self.items():
+                self.addItem(self.vline)
+            if self.hline not in self.items():
+                self.addItem(self.hline)
+            if self.crosshair not in self.items():
+                self.addItem(self.crosshair)
+        else:
+            if self.vline in self.items():
+                self.removeItem(self.vline)
+            if self.hline in self.items():
+                self.removeItem(self.hline)
+            if self.crosshair in self.items():
+                self.removeItem(self.crosshair)
+        return
+
+    def set_crosshair_pos(self, pos):
+        try:
+            pos = (pos[0], pos[1])
+        except TypeError:
+            pos = (pos.x(), pos.y())
+        size = self.crosshair.size()
+        self.crosshair.blockSignals(True)
+        self.vline.blockSignals(True)
+        self.hline.blockSignals(True)
+        self.crosshair.setPos(pos[0] - size[0] / 2, pos[1] - size[1] / 2)
+        self.vline.setPos(pos[0])
+        self.hline.setPos(pos[1])
+        self.crosshair.blockSignals(False)
+        self.vline.blockSignals(False)
+        self.hline.blockSignals(False)
+        self.sigCrosshairPosChanged.emit(QtCore.QPointF(*pos))
+        return
+
+    def set_crosshair_size(self, size):
+        try:
+            size = (size[0], size[1])
+        except TypeError:
+            size = (size.width(), size.height())
+        pos = self.vline.pos()
+        pos[1] = self.hline.pos()[1] - size[1] / 2
+        pos[0] -= size[0] / 2
+        self.crosshair.blockSignals(True)
+        self.crosshair.setSize(size)
+        self.crosshair.setPos(pos)
+        self.crosshair.blockSignals(False)
+        return
+
+    def set_crosshair_pen(self, pen):
+        self.crosshair.setPen(pen)
+        self.vline.setPen(pen)
+        self.hline.setPen(pen)
+        return
+
+class ScanViewBox(ViewBox):
+    """
+    Extension for pg.ViewBox to be used with ScanPlotWidget.
+
+    Implements optional rectangular rubber band area selection and optional corresponding zooming.
+    """
+
+    sigMouseAreaSelected = QtCore.Signal(QtCore.QRectF)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.zoom_by_selection = False
+        self.rectangle_selection = False
+        return
+
+    def toggle_selection(self, enable):
```

Comment:
Why are you using toggle and not just `@property` and `@xyz.setter`?

Reply:
For two reasons mainly:

- Since this is a derived Qt5 widget I felt like it is better practice to stick with Qt-like implementations with function calls rather than pythonic. I just introduced some properties for read-only purposes (no setters). This is also what you would expect when using (py)Qt-native widgets.
- As Qt widget it is quite common to connect signals to this kind of methods. If you want to use the method as Qt slot, it's more convenient.

Reply:
OK, I was genuinly just wondering.

At some point in the future (and when pyqtgraph gets there shit in order) we shoud look at best practices, but unifying the style is really something for the future.